### PR TITLE
Introduce automatic module names into all libraries being built.

### LINKED ISSUE #691: Support JDK 9 modules
Refer to http://branchandbound.net/blog/java/2017/12/automatic-module-name/ for useful background.

Our issue at Microsoft is that our JDK 9-supporting libraries have a dependency on opentelemetry, but our build gives us a warning that we have a dependency on a library that does not have a module name specified. If the library could at the least have a `Automatic-Module-Name`, that would be hugely beneficial for us. Going further than this, to have a module-info.java is unnecessary, but a nice to have.

Adding this would be extremely quick and easy, and I would really appreciate it if this could be considered for a release in the near future.

### PR DESCRIPTION
**Pull request by JonathanGiles**
This is a proposal to fix #691, by adding automatic module names into the built manifest.mf files that ship within the jar files for each library. These names are all proposals and I am happy to change them as you suggest. Thanks.

Fixes #691

### CODE REVIEW

**Comment by jkwatson** [comment]
Is there any gradle magic we can do so we don't have repeat that jar section in every build.gradle? Some stop level `allSubModules` thing, maybe?

**Comment by JonathanGiles** [comment]
I wondered the same thing. My Gradle these days is weak, so I was hoping someone else had guidance, otherwise I'll go and research.

**Comment by carlosalberto** [comment]
It should be possible to re-use this part, although I don't remember myself now the specifics. Once you do this, we should be able to merge this PR.

**Comment by jkwatson** [comment]
Also, please sign the CLA. :) Thanks very much for the contribution!

**Comment by JonathanGiles** [comment]
I did sign the CLA today, not sure how long it takes to update...

**Comment by jkwatson** [comment]
You might need to add a comment saying "I signed it"

**Comment by JonathanGiles** [comment]
I signed it

**Comment by jkwatson** [comment]
Guess not. Sometimes that kicks the CLAbot into action

File: api/build.gradle [hunk 1]
```diff
@@ -12,3 +13,11 @@ dependencies {
...
     exclude 'io/opentelemetry/internal/**'
 }
+
+jar {
+    inputs.property("moduleName", moduleName)
+
+    manifest {
+        attributes  'Automatic-Module-Name': moduleName
+    }
+}
```
Comment (conniey) [inline api/build.gradle]: Missing empty newline at the end of these.

**Comment by JonathanGiles** [comment]
I think we should centralise it but still explicitly set the module names. From my experience doing it as you suggest tends to lead to... Surprises after it is too late. But happy to follow consensus...

**Comment by thisthat** [comment]
With a quick test, I see that there is some difference between package structure and module name. Therefore, I agree with you that setting explicitly the module names is a better strategy on the long term.

File: api/build.gradle (hunk 1 above)
Comment (bogdandrutu) [reply api/build.gradle]: Please fix this everywhere.

**Comment by JonathanGiles** [comment]
I managed to centralise the configuration into the root build.gradle. Please take another look and let me know of any further concerns. Thanks.

File: api/build.gradle (hunk 1 above)
Comment (JonathanGiles) [reply api/build.gradle]: I've added an empty new line to the root build.gradle, as it did not have one (prior to my involvement). All other files no longer have been touched at the end of the file, so they remain as they were.

**Comment by carlosalberto** [comment]
LGTM.

File: api/build.gradle [hunk 2]
```diff
@@ -1,4 +1,5 @@
 description = 'OpenTelemetry API'
+ext.moduleName = "io.opentelemetry.api"
```
Comment (bogdandrutu) [inline api/build.gradle]: we don't have "api" in the package names, does that matter?

Comment (JonathanGiles) [reply api/build.gradle]: No, it doesn't matter. Module names do not need to relate to the package name, although they are a good starting point for choosing a good module name.

**Pull request merged**